crates/store/re_chunk_store/src/dataframe.rs

```diff
@@ -0,0 +1,499 @@
+            Self::Time(descr) => descr.to_arrow_field(),
+            Self::Component(descr) => descr.to_arrow_field(),
+        }
+    }
+}
+
+/// Describes a column used to control Rerun's behavior, such as `RowId`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ControlColumnDescriptor {
+    /// Semantic name associated with this data.
+    ///
+    /// Example: `rerun.controls.RowId`.
+    pub component_name: ComponentName,
+
+    /// The Arrow datatype of the column.
+    pub datatype: ArrowDatatype,
+}
+
+impl PartialOrd for ControlColumnDescriptor {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for ControlColumnDescriptor {
+    #[inline]
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        let Self {
+            component_name,
+            datatype: _,
+        } = self;
+        component_name.cmp(&other.component_name)
+    }
+}
+
+impl ControlColumnDescriptor {
+    #[inline]
+    pub fn to_arrow_field(&self) -> ArrowField {
+        let Self {
+            component_name,
+            datatype,
+        } = self;
+
+        ArrowField::new(
+            component_name.to_string(),
+            datatype.clone(),
+            false, /* nullable */
+        )
+    }
+}
+
+/// Describes a time column, such as `log_time`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TimeColumnDescriptor {
+    /// The timeline this column is associated with.
+    pub timeline: Timeline,
+
+    /// The Arrow datatype of the column.
+    pub datatype: ArrowDatatype,
+}
+
+impl PartialOrd for TimeColumnDescriptor {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for TimeColumnDescriptor {
+    #[inline]
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        let Self {
+            timeline,
+            datatype: _,
+        } = self;
+        timeline.cmp(&other.timeline)
+    }
+}
+
+impl TimeColumnDescriptor {
+    #[inline]
+    pub fn to_arrow_field(&self) -> ArrowField {
+        let Self { timeline, datatype } = self;
+        ArrowField::new(
+            timeline.name().to_string(),
+            datatype.clone(),
+            false, /* nullable */
+        )
+    }
+}
+
+/// Describes a data/component column, such as `Position3D`.
+//
+// TODO(#6889): Fully sorbetize this thing? `ArchetypeName` and such don't make sense in that
+// context. And whatever `archetype_field_name` ends up being, it needs interning.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ComponentColumnDescriptor {
+    /// The path of the entity.
+    pub entity_path: EntityPath,
+
+    /// Optional name of the `Archetype` associated with this data.
+    ///
+    /// `None` if the data wasn't logged through an archetype.
+    ///
+    /// Example: `rerun.archetypes.Points3D`.
+    pub archetype_name: Option<ArchetypeName>,
```

Comment:
Do we have a clear plan to populate this in the short term? best effort if no ambiguity based on the indicator component? but that would require accessing codegen'd reflection from `re_dataframe`, right?

Or should we drop this until we have tagged components?

Reply:
> Do we have a clear plan to populate this in the short term? best effort if no ambiguity based on the indicator component? but that would require accessing codegen'd reflection from `re_dataframe`, right?

I investigated such shenanigans and quickly came to the conclusion that the pain wasn't worth it.

> Or should we drop this until we have tagged components?

That is where my investigation led me :upside_down_face: